### ISSUE TITLE
Training on custom dataset

### ISSUE THREAD

**Issue by maryam95hallal** [visionml/pytracking#289]
Thanks for your amazing work!
after extensive search I found that your framework is one of the best in computer vision object tracking.
everything is very neat and organized. and the source code is not that complicated. moreover installing and running pytracking is not that hard. nearly every other Repo using ROI pooling is very hard  to install and run. 
my question is about training on custom dataset.
after I have read the readme and the issues I didn't find much relevant information. I have found that I Must run the following script to prepare the data.` create_distractor_dataset.py`  [https://github.com/visionml/pytracking/blob/master/ltr/train_settings/dimp/super_dimp_simple.py](url).
I know that I should provide the paths of training and datasets in `local.py` but everything is a bit hazy.
please provide a detailed instructions about how to prepare my custom dataset for training.
looking forward to your reply.
Best regards.

**Comment by goutamgmb** [comment]
Hi,

Thanks for the appreciation!

Which dataset do you want to integrate? Is it your own custom dataset, or do you want to use the distractor dataset used in our recent "KeepTrack" tracker?

Regards
Goutam

**Comment by maryam95hallal** [comment]
Hey,
I want to train trackers on my own custom dataset. I using "KeepTrack" tracker as well as other trackers.
Regards.

**Comment by maryam95hallal** [comment]
do I need to apply special configuration for keep tracking?

**Comment by maryam95hallal** [comment]
@goutamgmb  So my team and I are working on creating a new benchamrk dataset and we are using the known computer vision task and the open source project to compare the results of each tracker between different benchmarks. our benchmark is on development. so we need test each tracker on our dataset

**Comment by 2006pmach** [comment]
In this case wouldn't it be easier to just use the pre-trained KeepTrack model and simply test it on your new dataset? The tracker is already trained on four standard datasets (Got10k, LaSOT, Trackingnet, MSCOCO) in order to generalize well even to new datasets. Just check [here](https://github.com/visionml/pytracking/tree/master/pytracking/evaluation) how other datasets are integrated in our framework for example [LasotExtSub](https://github.com/visionml/pytracking/blob/master/pytracking/evaluation/lasotextensionsubsetdataset.py).

**Comment by maryam95hallal** [comment]
@2006pmach yeah applying the pre trained weights is easier yet we are aiming to test the full potential of the tracker. and to test our datasets quality compared to the other benchmarks. the old benchmarks have some flows we are trying to build a better benchmark test that could test the maximum potential of the trackers. as you know the data is a very important aspect of the project. we aim to derive weights of the model from a clean dataset so we can test the limits of each tracker and if we successfully addressed the problems we are targeting.

**Comment by maryam95hallal** [comment]
@2006pmach @goutamgmb 
one of our techniques used we are including objects similar to the target to be tracked without labeling so that the tracker learns the key differences between those two close but different objects.
I'm not authorized to further discuss the innovation of our benchmarks. yet if you could provide the help needed to integrate new dataset to that framework we will finish sooner thus the benchmark will be available sooner.
Regards.

**Comment by martin-danelljan** [comment]
Hi. Happy to see that you want to integrate your datasets into pytracking. You will find plenty of examples how to integrate your training dataset. It is not difficult. Just follow, for example, how we integrated the GOT10k or LaSOT datasets.

After your new dataset and benchmark paper is done, we would of course be excited if you also do a PR to pytracking!

Regards,
Martin

**Comment by 123wfl** [comment]
How to train lasot dataset with keep_track or can you provide the json file of lasot
